extensions/amp-consent/0.1/test/test-consent-config.js

```diff
@@ -556,28 +584,61 @@ describes.realWin('ConsentConfig', {amp: 1}, (env) => {
           // RANDOM is not allowed
           'r=RANDOM'
       );
+
       expect(url).to.match(
         /cid=amp-.{22}&pid=[0-9]+&pid64=.{22}&sourceurl=about%3Asrcdoc&r=RANDOM/
       );
     });
 
     it('override CLIENT_ID scope', async () => {
+      const config = dict({
+        'consentInstanceId': 'test',
+        'checkConsentHref': '/override',
+        'consentRequired': true,
+        'clientConfig': {
+          'test': 'ABC',
+        },
+        'postPromptUI': 'test',
+      });
+      const consentElement = createConsentElement(doc, config);
+      doc.body.appendChild(consentElement);
+      const fakeConsentStateManager = new ConsentStateManager();
+      fakeConsentStateManager.ampdoc_ = doc.body;
+
+      const stubGetServicePromiseForDoc = env.sandbox.stub(
+        GetService,
+        'getServicePromiseForDoc'
+      );
+
+      stubGetServicePromiseForDoc
+        .withArgs(env.sandbox.match.any, 'consentStateManager')
+        .resolves(fakeConsentStateManager);
+
+      stubGetServicePromiseForDoc.callThrough();
+
       const u1 = await expandConsentEndpointUrl(
         doc.body,
-        'https://example.test?cid=CLIENT_ID'
+        'https://example.test?cid=CLIENT_ID&pid=PAGE_VIEW_ID&clientconfig=CONSENT_INFO(clientConfig)&cpid='
       );
+
       const u2 = await expandConsentEndpointUrl(
         doc.body,
-        'https://example.test?cid=CLIENT_ID()'
+        'https://example.test?cid=CLIENT_ID()&pid=PAGE_VIEW_ID&clientconfig=CONSENT_INFO(clientConfig)&cpid='
       );
+
       const u3 = await expandConsentEndpointUrl(
         doc.body,
-        'https://example.test?cid=CLIENT_ID(123)'
+        'https://example.test?cid=CLIENT_ID(123)&pid=PAGE_VIEW_ID&clientconfig=CONSENT_INFO(clientConfig)&cpid='
       );
+
       const u4 = await expandConsentEndpointUrl(
         doc.body,
-        'https://example.test?cid=CLIENT_ID(abc)'
+        'https://example.test?cid=CLIENT_ID(abc)&pid=PAGE_VIEW_ID&clientconfig=CONSENT_INFO(clientConfig)&cpid='
       );
+
+      await macroTask();
+      await macroTask();
```

Comment:
Why twice?

Reply:
It had not worked with only one `macroTask()` on previous implementation but it seems to be working fine now. Fixed.